hs_geographic_feature_resource/models.py

```diff
@@ -0,0 +1,202 @@
+        # See Shapefile format: http://resources.arcgis.com/en/help/main/10.2/index.html#//005600000003000000
+        return (".zip", ".shp", ".shx", ".dbf", ".prj", ".sbx", ".sbn", ".cpg", ".xml", ".fbn", ".fbx", ".ain",
+                ".aih", ".atx", ".ixs", ".mxs")
+
+    @classmethod
+    def can_have_multiple_files(cls):
+        # can have more than one files
+        return True
+
+    class Meta:
+        verbose_name = 'Geographic Feature (ESRI Shapefiles)'
+        proxy = True
+
+processor_for(GeographicFeatureResource)(resource_processor)
+
+# define the GeographicFeatureMetaData metadata
+class GeographicFeatureMetaData(CoreMetaData):
+    geometryinformation = generic.GenericRelation(GeometryInformation)
+    fieldinformation = generic.GenericRelation(FieldInformation)
+    originalcoverage = generic.GenericRelation(OriginalCoverage)
+    originalfileinfo = generic.GenericRelation(OriginalFileInfo)
+
+    @classmethod
+    def get_supported_element_names(cls):
+        # get the names of all core metadata elements
+        elements = super(GeographicFeatureMetaData, cls).get_supported_element_names()
+        # add the name of any additional element to the list
+        elements.append('FieldInformation')
+        elements.append('OriginalCoverage')
+        elements.append('GeometryInformation')
+        elements.append('OriginalFileInfo')
+        return elements
+
+    def has_all_required_elements(self):
+        if self.get_required_missing_elements():
+            return False
+        return True
+
+    def get_required_missing_elements(self):  # show missing required meta
+        missing_required_elements = super(GeographicFeatureMetaData, self).get_required_missing_elements()
+        if not (self.coverages.all().filter(type='box').first() or self.coverages.all().filter(type='point').first()):
+            missing_required_elements.append('Spatial Coverage')
+        if not self.originalcoverage.all().first():
+            missing_required_elements.append('Spatial Reference')
+        if not self.geometryinformation.all().first():
+            missing_required_elements.append('Geometry Information')
+        if not self.originalfileinfo.all().first():
+            missing_required_elements.append('Resource File Information')
+
+        return missing_required_elements
+
+    def get_xml(self):
+        from lxml import etree
+        # get the xml string representation of the core metadata elements
+        xml_string = super(GeographicFeatureMetaData, self).get_xml(pretty_print=False)
+
+        # create an etree xml object
+        RDF_ROOT = etree.fromstring(xml_string)
+
+        # get root 'Description' element that contains all other elements
+        container = RDF_ROOT.find('rdf:Description', namespaces=self.NAMESPACES)
+
+        if self.originalcoverage.all().first():
+            hsterms_ori_cov = etree.SubElement(container, '{%s}spatialReference' % self.NAMESPACES['hsterms'])
```

Comment:
@zhiyuli Can you refactor to use add_metadata_element_to_xml() to write your elements to XML?  This is used in [hs_geo_raster_resource](https://github.com/hydroshare/hydroshare/blob/6eeec23d1a631de0b064f54fd862e84f14368e2f/hs_geo_raster_resource/models.py#L330), and will drastically reduce the amount of code in your get_xml() method.